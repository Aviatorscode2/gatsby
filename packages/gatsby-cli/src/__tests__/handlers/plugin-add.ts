import { copyFile, readFile, rm } from "fs/promises"
import { resolve } from "path"
import { addPlugins } from "../../handlers/plugin-add"

/**
 * Copy files from minimal starters instead of testing against static gatsby-configs
 * in fixtues so that these break if we change the starter configs in the future.
 */

const fixtures = resolve(`packages/gatsby-cli/src/__tests__/fixtures`)
const config = {
  starter: resolve(`starters/gatsby-starter-minimal/gatsby-config`),
  fixture: `${fixtures}/gatsby-config`,
}
const plugin = {
  hello: `gatsby-plugin-hello`,
  world: `gatsby-plugin-world`,
}

describe(`installPlugins write to gatsby-config.js`, () => {
  beforeEach(async () => {
    await copyFile(
      resolve(`${config.starter}.js`),
      resolve(`${config.fixture}.js`)
    )
  })

  afterEach(async () => {
    await rm(resolve(`${config.fixture}.js`))
  })

  it(`should not write with no plugins`, async () => {
    await addPlugins([], {}, fixtures, [])
    const gatsbyConfig = (await readFile(`${config.fixture}.js`)).toString()
    expect(gatsbyConfig).toMatchSnapshot()
  })

  it(`should write a single plugin`, async () => {
    await addPlugins([plugin.hello], {}, fixtures, [])
    const gatsbyConfig = (await readFile(`${config.fixture}.js`)).toString()
    expect(gatsbyConfig).toMatchSnapshot()
  })

  it(`should write multiple plugins`, async () => {
    await addPlugins([plugin.hello, plugin.world], {}, fixtures, [])
    const gatsbyConfig = (await readFile(`${config.fixture}.js`)).toString()
    expect(gatsbyConfig).toMatchSnapshot()
  })

  it(`should write a plugin with options`, async () => {
    await addPlugins(
      [plugin.hello],
      {
        [plugin.hello]: {
          greet: true,
        },
      },
      fixtures,
      []
    )
    const gatsbyConfig = (await readFile(`${config.fixture}.js`)).toString()
    expect(gatsbyConfig).toMatchSnapshot()
  })
})
