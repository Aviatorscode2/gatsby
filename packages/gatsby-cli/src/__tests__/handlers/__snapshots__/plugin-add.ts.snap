// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installPlugins write to gatsby-config.js should not write with no plugins 1`] = `
"module.exports = {
  siteMetadata: {
    siteUrl: \`https://www.yourdomain.tld\`,
  },
  plugins: [],
}
"
`;

exports[`installPlugins write to gatsby-config.js should write a plugin with options 1`] = `
"module.exports = {
  siteMetadata: {
    siteUrl: \`https://www.yourdomain.tld\`
  },
  plugins: [{
    resolve: 'gatsby-plugin-hello',
    options: {
      \\"greet\\": true
    }
  }]
};"
`;

exports[`installPlugins write to gatsby-config.js should write a single plugin 1`] = `
"module.exports = {
  siteMetadata: {
    siteUrl: \`https://www.yourdomain.tld\`
  },
  plugins: [\\"gatsby-plugin-hello\\"]
};"
`;

exports[`installPlugins write to gatsby-config.js should write multiple plugins 1`] = `
"module.exports = {
  siteMetadata: {
    siteUrl: \`https://www.yourdomain.tld\`
  },
  plugins: [\\"gatsby-plugin-hello\\", \\"gatsby-plugin-world\\"]
};"
`;
